classification.py

```diff
@@ -14,32 +14,43 @@
 )
 parser.add_argument("--dataset", choices=("solidletters",), help="Dataset to train on")
 parser.add_argument("--dataset_path", type=str, help="Path to dataset")
-parser.add_argument("--epochs", type=int, default=100, help="Number of epochs to train")
 parser.add_argument("--batch_size", type=int, default=64, help="Batch size")
-parser.add_argument("--cpu", action="store_true", help="Use the CPU for training/testing")
+parser.add_argument(
+    "--num_workers",
+    type=int,
+    default=0,
+    help="Number of workers for the dataloader. NOTE: set this to 0 on Windows, any other value leads to poor performance",
+)
 parser.add_argument(
     "--checkpoint",
     type=str,
     default=None,
     help="Checkpoint file to load weights from for testing",
 )
+parser.add_argument(
```

Comment:
Setting the experiment name is good.

My recommendation would be to have sub-dirs inside the experiment directory, perhaps something like this 

```
./results/$experiment_name/$month_day/$hour_min_second
```
It's much easier to find things by date and time than by the version number which lightning generates by default.   If the user doesn't specify an experiment name then the output could be written to 

```
./results/$month_day/$hour_min_second
```

BRepNet writes logs to
```
{opts.log_dir}/{month_day}/{hour_min_second}
```
which gives folders which look like this
```
    logs/0430/123103/checkpoints
    logs/0430/123103/hparams.yaml
    logs/0430/123103/events.out.tfevents.xxxx
```
I think I like the idea of having a user defined `experiment_name` and will add that to BRepNet myself.

It's also nice to echo the log dir to the shell at the start and end of training.

Reply:
I incorporated all these suggestions, thank you.